mdBook 0.5 support

There was an [major release](https://github.com/rust-lang/mdBook/releases/tag/v0.5.0) of mdbook today. This preprocessor is incompatible with the new version.
```
 INFO Book building has started
[2025-11-18T14:52:40Z ERROR mdbook_admonish] Fatal error: Unable to parse the input
[2025-11-18T14:52:40Z ERROR mdbook_admonish]   - Unable to parse the input
[2025-11-18T14:52:40Z ERROR mdbook_admonish]   - invalid type: null, expected any valid TOML value at line 1 column 147
 WARN Error writing the RenderContext to the backend, Broken pipe (os error 32)
ERROR The "admonish" preprocessor exited unsuccessfully with exit status: 1 status
```

Regarding whether it works with mdbook binary `0.4.x`, at least in its current state, it will result in an ERROR.

```
2025-11-24 12:40:20 [INFO] (mdbook::book): Book building has started
[ERROR mdbook_admonish] Fatal error: Unable to parse the input
[ERROR mdbook_admonish]   - Unable to parse the input
[ERROR mdbook_admonish]   - missing field `items` at line 1 column 841446
2025-11-24 12:40:20 [ERROR] (mdbook::utils): Error: The "admonish" preprocessor exited unsuccessfully with exit status: 1 status
```

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -31,7 +31,7 @@ jobs:
       - name: Install mdbook & plugins
         uses: taiki-e/install-action@v2.18.6
         with:
-          tool: mdbook@0.5.3,mdbook-admonish@1.20.0,mdbook-alerts@0.8.0,mdbook-emojicodes@0.3.0
+          tool: mdbook@0.4.52,mdbook-admonish@1.20.0,mdbook-alerts@0.8.0,mdbook-emojicodes@0.3.0
       - name: Update SUMMARY.md with proposals
         run: python3 tools/auto-update-design-proposals.py
       - name: Setup Pages

